spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+
+    describe 'POST #create' do
+      context "tickets are available and user hasn't bought any" do
+        before do
+          create(:ticket, conference: conference_with_open_registration)
+          post :create, registration: attributes_for(:registration),
+                        conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns user variable' do
+          expect(assigns(:user)).to eq user
+        end
+
+        it 'redirects to conference tickets path' do
+          expect(response).to redirect_to conference_tickets_path(conference_with_open_registration.short_title)
+        end
+
+        it 'creates a new registration' do
+          expect(Registration.count).to eq 1
+        end
+      end
+
+      context 'registration save fails' do
+        before do
+          allow_any_instance_of(Registration).to receive(:save).and_return(false)
+          post :create, registration: attributes_for(:registration),
+                        conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'renders the new template' do
+          expect(response).to render_template('new')
+        end
+
+        it 'does not create registration' do
+          expect(Registration.count).to eq 0
+        end
+      end
+    end
+
+    describe 'PATCH #update' do
+      before do
+        @registration = create(:registration,
+                               conference: conference_with_open_registration,
+                               user: user,
+                               arrival: Date.new(2014, 04, 25))
+      end
+
+      context 'updates successfully' do
+        before do
+          patch :update, registration: attributes_for(:registration, arrival: Date.new(2014, 04, 29)),
+                         conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to registration show path' do
+          expect(response).to redirect_to conference_conference_registrations_path(conference_with_open_registration.short_title)
+        end
+
+        it 'shows success message in flash notice' do
+          expect(flash[:notice]).to match('Registration was successfully updated.')
+        end
+
+        it 'updates the registration' do
+          @registration.reload
+          expect(@registration.arrival).to eq Date.new(2014, 04, 29)
+        end
+      end
+
+      context 'update fails' do
+        before do
+          allow_any_instance_of(Registration).to receive(:update_attributes).and_return(false)
+          patch :update, registration: attributes_for(:registration, arrival: Date.new(2014, 04, 27)),
+                         conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'renders edit template' do
+          expect(response).to render_template('edit')
+        end
+
+        it 'shows error in flash message' do
+          expect(flash[:error]).to include 'Could not update your registration'
```

Comment:
Your error message must include this part of the code too: `for #{@conference.title}:`